### PR TITLE
Fix message storing

### PR DESCRIPTION
**Pull request by fisyamaloy**
- when user click "send", his message stores in appropriate tables of Postgre Repository.
- added struct of channelInfo. Now client-side stores channels with not only channels names, but with their IDs and channels creators.
- many functions of postgre repository was changed.
- added new documenation for storing message and his codes

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,7 +14,7 @@ set(EXPORT_COMPILE_COMMANDS TRUE)
 # GCC TBD
 if (MSVC)
     # warning level 4 and all warnings as errors
-    add_compile_options(/W4 /WX)
+    add_compile_options(/W3 /WX)
```
Comment (f1nal3) [inline CMakeLists.txt:17]: How does that correspond to this PR name? Why should it be changed?

File: DataAccess/DataAccess.Postgre/DataBaseOptions.hpp [hunk 2]
```diff
@@ -4,7 +4,7 @@
 namespace DBOptions
 {
     constexpr std::string_view real = 
-        "dbname=juniorgram user=postgres hostaddr=127.0.0.1 port=5432";
```
Comment (f1nal3) [inline DataAccess/DataAccess.Postgre/DataBaseOptions.hpp:7]: Was there any change to our database naming? Why did it had to happen? Currently that DB is empty:
image

File: Network/Public/Include/Network/Primitives.hpp [hunk 3]
```diff
@@ -113,4 +113,30 @@ namespace Network
         ar& o.userId& o.messageId;
     }
 
+    struct MessageStoringInfo : MessageInfo
```
Comment (f1nal3) [inline Network/Public/Include/Network/Primitives.hpp:116]: Do we use the old message info anywhere? Is there a reason to keep the old one?

File: Server/Server.Exec/MainServer.cpp [hunk 4]
```diff
@@ -8,14 +8,14 @@
 #include <Server.hpp>
 #include <ArgumentParser.hpp>
 
-int main(int argc, const char** argv)
+int main(/*int argc, const char** argv*/)
 {
 
     try
```
Comment (f1nal3) [inline Server/Server.Exec/MainServer.cpp:14]: What is the motivation to changing MainServer.cpp?

**Review by f1nal3 — CHANGES_REQUESTED** [review]

**Comment by f1nal3** [comment]
K, never mind the review, didn't notice it's a draft :D

**Comment by fisyamaloy** [comment]
> K, never mind the review, didn't notice it's a draft :D

xD

**Review by vDrWool — APPROVED** [review]

**Review by GuskoMaksim — APPROVED** [review]

**Pull request merged**